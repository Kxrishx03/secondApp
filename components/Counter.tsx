import React from 'react';
import {StyleSheet, Text, Pressable, View} from 'react-native';
import {Colors} from 'react-native/Libraries/NewAppScreen';
import useCounter from '@/hooks/useCounter';

export function Counter(){
  const {count, increment, decrement} = useCounter();

  return (
    <View style={styles.body}>
      <View style={styles.sectionContainer}>
        <Text accessibilityRole='header' style={styles.sectionTitle}>Current count: {count}</Text>
        <Pressable accessibilityRole='button' style={styles.button} onPress={decrement}>
          <Text testID='decrement'>Decrement</Text>
        </Pressable>
        <Pressable accessibilityRole='button' style={styles.button} onPress={increment}>
          <Text testID='increment'>Increment</Text>
        </Pressable>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  body: {
    backgroundColor: Colors.white,
  },
  sectionContainer: {
    padding: 24,
    alignItems: 'center',
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
    color: Colors.black,
  },
  button: {
    borderRadius: 12,
    padding: 6,
    margin: 6,
    backgroundColor: '#9e9ef8',
    justifyContent: 'center',
    alignItems: 'center',
  },
});