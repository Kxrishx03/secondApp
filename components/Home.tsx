import { useRouter } from "expo-router";
import { Button, Text, View } from "react-native";

export default function Home() {
  const {push} = useRouter();

  return (
    <View>
      <Text accessibilityRole="header">Home Screen</Text>
      <Button
        accessibilityLabel="Navigate to Profile Screen"
        title="Go to Profile"
        onPress={() => push("/profile")} 
      />
    </View>
  );
}
