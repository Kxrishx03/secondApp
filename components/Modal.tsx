import React from 'react';
import { View, Text, Button } from 'react-native';
import { useRouter } from 'expo-router';

export default function Modal() {
  const router = useRouter();

  const handleConfirm = () => {
    // Logic to handle confirmation
    console.log('Action confirmed!');
    router.back();
  };

  return (
    <View style={{ padding: 20 }}>
      <Text style={{ fontSize: 24 }}>This is a Modal</Text>
      <Button title="Confirm" onPress={handleConfirm} />
      <Button title="Cancel" onPress={() => router.back()} />
    </View>
  );
}