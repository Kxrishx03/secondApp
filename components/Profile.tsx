import { View, Text } from 'react-native';
import React from 'react';
import { useLocalSearchParams } from 'expo-router';

export default function () {
    const { id } = useLocalSearchParams<{ id?: string }>()
    return (
        <View>
            <Text testID='profile-id'>{id ? `Profile ID: ${id}` : 'No Profile ID provided'}</Text>
        </View>

    );
  
}