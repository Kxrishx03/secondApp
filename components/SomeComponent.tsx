import React from 'react';
import { View, Text, Button } from 'react-native';
import { useRouter } from 'expo-router';

const SomeComponent = () => {
  const router = useRouter();

  const handleNavigation = async () => {
    await new Promise(resolve => setTimeout(resolve, 1000));
    router.push('/nextPage');
  };

  return (
    <View>
      <Text accessibilityRole='header'>Current Page</Text>
      <Button accessibilityLabel='Next-page-btn' title="Go to Next Page" onPress={handleNavigation} />
    </View>
  );
};

export default SomeComponent;