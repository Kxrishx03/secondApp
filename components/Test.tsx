import { View } from "react-native";

export  function TestScreen() {
  return (
    
      <View testID="test-view">
        Testing Testing ...Hiii....
      </View>
  );
}
