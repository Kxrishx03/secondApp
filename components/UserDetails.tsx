// src/components/UserDetail.tsx
import React from 'react';
import { View, Text, Button } from 'react-native';
import { useLocalSearchParams, useRouter } from 'expo-router';

interface UserDetailParams {
    userId?: string;
    showDetails?: string; 
}

const UserDetail: React.FC = () => {
    const { userId, showDetails } = useLocalSearchParams<UserDetailParams>();
    const router = useRouter();

    const handleNavigate = () => {
        router.push('/user');
    };

    return (
        <View>
            <Text accessibilityLabel='userId'>{userId ? `User  ID: ${userId}` : 'No User ID provided'}</Text>
            {showDetails === 'true' && <Text accessibilityRole='header'>Showing detailed information...</Text>}
            <Button accessibilityLabel='edit-user' title="Edit User" onPress={handleNavigate} />
        </View>
    );
};

export default UserDetail;