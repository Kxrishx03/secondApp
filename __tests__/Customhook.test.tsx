import React from 'react'; 
import { Counter } from '@/components/Counter';
import { render ,screen, fireEvent, renderHook, act} from '@testing-library/react-native';
import useCounter from '@/hooks/useCounter';
import renderer from 'react-test-renderer';

describe('Custom Hook Testing',() =>{

    //Integration method
    test('Integrtion methods',()=>{
          render(<Counter />);
         const decrement = screen.getByRole('button',{name:'Decrement'});
         const increment = screen.getByRole('button',{name:'Increment'});

         //Before starting of event 
         expect(screen.getByRole('header',{name:'Current count: 0'})).toBeOnTheScreen();
         
         //After increment
         fireEvent.press(increment);
         expect(screen.getByRole('header',{name:'Current count: 1'})).toBeOnTheScreen();

         //After decremnet
         fireEvent.press(decrement);
         expect(screen.getByRole('header',{name:'Current count: 0'})).toBeOnTheScreen();

    });

    //Isolation Testing
    test('Isolation testing',()=>{
      const {result} = renderHook(useCounter);
      expect(result.current.count).toBe(0);
      //Increment
      act(()=>result.current.increment()); 
      expect(result.current.count).toBe(1);
      //Decrement
      act(()=>result.current.decrement()); 
      expect(result.current.count).toBe(0);
    });

    //Snapshot Testing
    test('Snapshot practice',()=>{
      const tree = renderer.create(<Counter />).toJSON();
      expect(tree).toMatchSnapshot();
    })
});