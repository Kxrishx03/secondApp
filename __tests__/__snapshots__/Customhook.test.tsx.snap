// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Hook Testing Snapshot practice 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFF",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "padding": 24,
      }
    }
  >
    <Text
      accessibilityRole="header"
      style={
        {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "600",
        }
      }
    >
      Current count: 
      0
    </Text>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#9e9ef8",
          "borderRadius": 12,
          "justifyContent": "center",
          "margin": 6,
          "padding": 6,
        }
      }
    >
      <Text
        testID="decrement"
      >
        Decrement
      </Text>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#9e9ef8",
          "borderRadius": 12,
          "justifyContent": "center",
          "margin": 6,
          "padding": 6,
        }
      }
    >
      <Text
        testID="increment"
      >
        Increment
      </Text>
    </View>
  </View>
</View>
`;
